src/Compilers/Core/Portable/WellKnownMembers.cs

```diff
@@ -2681,13 +2687,61 @@ static WellKnownMembers()
                 0,                                                                                                          // Arity
                     (byte)SignatureTypeCode.GenericTypeParameter, 6,                                                        // Field Signature
 
+                // System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest__Item1
+                (byte)MemberFlags.Field,                                                                                    // Flags
+                (byte)WellKnownType.ExtendedType, WellKnownType.System_Runtime_CompilerServices_ValueTuple_T1_T2_T3_T4_T5_T6_T7_TRest.ExtensionByte(), // DeclaringTypeId
```

Comment:
You're correct. There was an impact on the generated IL for the constructor. 
But it seems that most bytes are still stored statically. Only a few are produced at runtime by the compiler with `ExtensionByte`.
I will try the same thing replacing ExtensionByte with (byte)(x-255) to compare the result.

Before:
```
.method private hidebysig specialname rtspecialname static 
	void .cctor () cil managed 
{
	// Method begins at RVA 0x41a68
	// Code size 3187 (0xc73)
	.maxstack 4
	.locals init (
		[0] uint8[],
		[1] string[]
	)

	IL_0000: ldc.i4 2700
	IL_0005: newarr [System.Runtime]System.Byte
	IL_000a: dup
	IL_000b: ldtoken field valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=2700' '<PrivateImplementationDetails>'::EC51BC739CB9D330F73835E6C77AB1F5BF232A31
	IL_0010: call void [System.Runtime]System.Runtime.CompilerServices.RuntimeHelpers::InitializeArray(class [System.Runtime]System.Array, valuetype [System.Runtime]System.RuntimeFieldHandle)
	IL_0015: stloc.0
	IL_0016: ldc.i4 294
	IL_001b: newarr [System.Runtime]System.String
	IL_0020: dup
	IL_0021: ldc.i4.0
	IL_0022: ldstr "Round"
	IL_0027: stelem.ref
	IL_0028: dup
	IL_0029: ldc.i4.1
	IL_002a: ldstr "Pow"
	IL_002f: stelem.ref
	IL_0030: dup
	IL_0031: ldc.i4.2
```



After:
```
.method private hidebysig specialname rtspecialname static 
	void .cctor () cil managed 
{
	// Method begins at RVA 0x57118
	// Code size 3773 (0xebd)
	.maxstack 4
	.locals init (
		[0] uint8[],
		[1] string[]
	)

	IL_0000: nop
	IL_0001: ldc.i4 2889
	IL_0006: newarr [System.Runtime]System.Byte
	IL_000b: dup
	IL_000c: ldtoken field valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=2889' '<PrivateImplementationDetails>'::'82FE9269DACA81A6359D20D1F2A5AF0913A6051B'
	IL_0011: call void [System.Runtime]System.Runtime.CompilerServices.RuntimeHelpers::InitializeArray(class [System.Runtime]System.Array, valuetype [System.Runtime]System.RuntimeFieldHandle)
	IL_0016: dup
	IL_0017: ldc.i4 2829
	IL_001c: ldc.i4 256
	IL_0021: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0026: stelem.i1
	IL_0027: dup
	IL_0028: ldc.i4 2835
	IL_002d: ldc.i4 256
	IL_0032: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0037: stelem.i1
	IL_0038: dup
	IL_0039: ldc.i4 2841
	IL_003e: ldc.i4 256
	IL_0043: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0048: stelem.i1
	IL_0049: dup
	IL_004a: ldc.i4 2847
	IL_004f: ldc.i4 256
	IL_0054: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_0059: stelem.i1
	IL_005a: dup
	IL_005b: ldc.i4 2853
	IL_0060: ldc.i4 256
	IL_0065: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_006a: stelem.i1
	IL_006b: dup
	IL_006c: ldc.i4 2859
	IL_0071: ldc.i4 256
	IL_0076: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_007b: stelem.i1
	IL_007c: dup
	IL_007d: ldc.i4 2865
	IL_0082: ldc.i4 256
	IL_0087: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_008c: stelem.i1
	IL_008d: dup
	IL_008e: ldc.i4 2871
	IL_0093: ldc.i4 256
	IL_0098: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_009d: stelem.i1
	IL_009e: dup
	IL_009f: ldc.i4 2883
	IL_00a4: ldc.i4 257
	IL_00a9: call uint8 Microsoft.CodeAnalysis.WellKnownTypeExtensions::ExtensionByte(valuetype Microsoft.CodeAnalysis.WellKnownType)
	IL_00ae: stelem.i1
	IL_00af: stloc.0
	IL_00b0: ldc.i4 330
	IL_00b5: newarr [System.Runtime]System.String
	IL_00ba: dup
	IL_00bb: ldc.i4.0
	IL_00bc: ldstr "Round"
	IL_00c1: stelem.ref
	IL_00c2: dup
	IL_00c3: ldc.i4.1
	IL_00c4: ldstr "Pow"
	IL_00c9: stelem.ref
	IL_00ca: dup
	IL_00cb: ldc.i4.2
```